src/controller.c

```diff
@@ -227,11 +230,109 @@ void eeprom_write(int block, const uint8_t * const data)
     };
     static unsigned long long output[8];
 
-	SI_eeprom_write_block[0] = 0x000000000a010500 | (block & 255);
-    memcpy( &SI_eeprom_write_block[1], data, 8 );
+    SI_eeprom_write_block[0] = 0x000000000a010500 | (block & 255);
+    memcpy( &SI_eeprom_write_block[1], data, EEPROM_BLOCK_SIZE );
     __controller_exec_PIF(SI_eeprom_write_block,output);
 }
 
+/**
+ * @brief Read a buffer of bytes from EEPROM
+ * 
+ * This is a convenience helper that abstracts away the 8-byte block access pattern.
+ * 
+ * @param[out] buf
+ *             Buffer to read data into
+ * @param[in]  start
+ *             Byte offset to start reading data from
+ * @param[in]  len
+ *             Byte length of data to read into buffer
+ */
+void eeprom_read_bytes(uint8_t * dest, size_t start, size_t len)
+{
+	uint8_t buf[EEPROM_BLOCK_SIZE];
+	size_t bytes_left = len;
+	size_t current_block = start / EEPROM_BLOCK_SIZE;
+	// If we need to read a partial block to start off...
+	size_t block_offset = start % EEPROM_BLOCK_SIZE;
+	if (block_offset)
+	{
+		eeprom_read(current_block++, buf);
+		bytes_left -= (EEPROM_BLOCK_SIZE - block_offset);
+		while (block_offset < EEPROM_BLOCK_SIZE)
+		{
+			*dest++ = buf[block_offset++];
+		}
+	}
+	// Read whole blocks at a time
+	while (bytes_left >= EEPROM_BLOCK_SIZE)
+	{
+		eeprom_read(current_block++, buf);
```

Comment:
We should be able to read directly into `dest` here as well IINM. Maybe GCC will already optimize it out, but I'm not sure if it serves readability as well